### ISSUE TITLE
PyTorch Engine throws "EngineException: Method 'forward' is not defined." when running Open AI Whisper zoo-model on GPU 

### ISSUE THREAD

**Issue by technologistkj** [deepjavalibrary/djl#2750]
## Description

When I run the openai whisper example from the DJL demo, on a GPU, I'm getting this error:

```
Caused by: ai.djl.engine.EngineException: Method 'forward' is not defined.
at ai.djl.pytorch.jni.PyTorchLibrary.moduleRunMethod(Native Method)
at ai.djl.pytorch.jni.IValueUtils.forward(IValueUtils.java:55)
at ai.djl.pytorch.engine.PtSymbolBlock.forwardInternal(PtSymbolBlock.java:145)
at ai.djl.nn.AbstractBaseBlock.forward(AbstractBaseBlock.java:79)
at ai.djl.nn.Block.forward(Block.java:127)
at ai.djl.inference.Predictor.predictInternal(Predictor.java:144)
at ai.djl.inference.Predictor.batchPredict(Predictor.java:171)
... 13 common frames omitted
```

### Expected Behavior

The inference was to be performed just as it was done in CPU mode.

### Error Message

```
Caused by: ai.djl.engine.EngineException: Method 'forward' is not defined.
at ai.djl.pytorch.jni.PyTorchLibrary.moduleRunMethod(Native Method)
at ai.djl.pytorch.jni.IValueUtils.forward(IValueUtils.java:55)
at ai.djl.pytorch.engine.PtSymbolBlock.forwardInternal(PtSymbolBlock.java:145)
at ai.djl.nn.AbstractBaseBlock.forward(AbstractBaseBlock.java:79)
at ai.djl.nn.Block.forward(Block.java:127)
at ai.djl.inference.Predictor.predictInternal(Predictor.java:144)
at ai.djl.inference.Predictor.batchPredict(Predictor.java:171)
... 13 common frames omitted
```

## How to Reproduce?

To reproduce this error, 

1. Avoiding found-two-devices error ("GPU and CPU found") by extending WhisperTranslator to WhisperGpuTranslator. In the extended class override the below function.

```    
//the input is moved to GPU, also placeholder is initialized with 0f instead of empty string to avoid the crash "SIGSEGV (0xb)" 
@Override
    public NDList processInput(TranslatorContext ctx, Audio input) throws Exception {
        NDArray samples = ctx.getNDManager().create(input.getData());
        for (AudioProcessor processor : processors) {
            samples = processor.extractFeatures(samples.getManager(), samples);
        }
        samples = samples.expandDims(0);
        NDArray placeholder = ctx.getNDManager().create(0f);
        placeholder.setName("module_method:generate");
        return new NDList(samples, placeholder).toDevice(device, true);
    }
```

- 

2. Extending WhisperTranslatorFactory to provide the above extended class

```
@AllArgsConstructor
public class WhisperGpuTranslatorFactory implements TranslatorFactory, Serializable {

    private static final long serialVersionUID = 1L;

    private Device device;

    private static final Set<Pair<Type, Type>> SUPPORTED_TYPES = new HashSet<>();

    static {
        SUPPORTED_TYPES.add(new Pair<>(Audio.class, String.class));
    }

    /** {@inheritDoc} */
    @Override
    public Set<Pair<Type, Type>> getSupportedTypes() {
        return SUPPORTED_TYPES;
    }

    /** {@inheritDoc} */
    @Override
    @SuppressWarnings("unchecked")
    public <I, O> Translator<I, O> newInstance(
            Class<I> input, Class<O> output, Model model, Map<String, ?> arguments) {
        if (input == Audio.class && output == String.class) {
            return (Translator<I, O>) new WhisperGpuTranslator(device);
        }
        throw new IllegalArgumentException("Unsupported input/output types.");
    }
}
```

3. Provide the above classes in the WhisperModel.java as the translator and translator factory respectively.

### Steps to reproduce

Ran the code which works when run on CPU:

```
Audio audioToRecognise = AudioFactory.newInstance().fromInputStream(new ByteArrayInputStream(wavBytes));
                transcript = asrModel.newPredictor().predict(audioToRecognise);
````

## What have you tried to solve it?

Tried these combinations of pytorch-native-cuda:
1. pytorch-native-cu111, 
2. pytorch-native-cu113, 
3. pytorch-native-cu116, 
3. pytorch-native-cu117, 
4. pytorch-native-cu118, 
5
## Environment Info

```
> Task :integration:debugEnv
[DEBUG] - Registering EngineProvider: XGBoost
[DEBUG] - Registering EngineProvider: LightGBM
[DEBUG] - Registering EngineProvider: OnnxRuntime
[DEBUG] - Registering EngineProvider: MXNet
[DEBUG] - Registering EngineProvider: PyTorch
[DEBUG] - Registering EngineProvider: TensorFlow
[DEBUG] - Found default engine: MXNet
----------- System Properties -----------
sun.desktop: gnome
awt.toolkit: sun.awt.X11.XToolkit
java.specification.version: 11
sun.cpu.isalist: 
sun.jnu.encoding: UTF-8
java.class.path: /home/dheeyantra/Projects/djl/integration/build/classes/java/main:/home/dheeyantra/Projects/djl/integration/build/resources/main:/home/dheeyantra/.gradle/caches/modules-2/files-2.1/commons-cli/commons-cli/1.5.0/dc98be5d5390230684a092589d70ea76a147925c/commons-cli-1.5.0.jar:/home/dheeyantra/.gradle/caches/modules-2/files-2.1/org.apache.logging.log4j/log4j-slf4j-impl/2.20.0/7ab4f082fd162f60afcaf2b8744a3d959feab3e8/log4j-slf4j-impl-2.20.0.jar:/home/dheeyantra/Projects/djl/basicdataset/build/libs/basicdataset-0.24.0-SNAPSHOT.jar:/home/dheeyantra/Projects/djl/model-zoo/build/libs/model-zoo-0.24.0-SNAPSHOT.jar:/home/dheeyantra/Projects/djl/testing/build/libs/testing-0.24.0-SNAPSHOT.jar:/home/dheeyantra/Projects/djl/engines/mxnet/mxnet-model-zoo/build/libs/mxnet-model-zoo-0.24.0-SNAPSHOT.jar:/home/dheeyantra/Projects/djl/engines/pytorch/pytorch-model-zoo/build/libs/pytorch-model-zoo-0.24.0-SNAPSHOT.jar:/home/dheeyantra/Projects/djl/engines/pytorch/pytorch-jni/build/libs/pytorch-jni-2.0.1-0.24.0-SNAPSHOT.jar:/home/dheeyantra/Projects/djl/engines/tensorflow/tensorflow-model-zoo/build/libs/tensorflow-model-zoo-0.24.0-SNAPSHOT.jar:/home/dheeyantra/Projects/djl/engines/ml/xgboost/build/libs/xgboost-0.24.0-SNAPSHOT.jar:/home/dheeyantra/Projects/djl/engines/ml/lightgbm/build/libs/lightgbm-0.24.0-SNAPSHOT.jar:/home/dheeyantra/Projects/djl/engines/onnxruntime/onnxruntime-engine/build/libs/onnxruntime-engine-0.24.0-SNAPSHOT.jar:/home/dheeyantra/.gradle/caches/modules-2/files-2.1/org.apache.logging.log4j/log4j-core/2.20.0/eb2a9a47b1396e00b5eee1264296729a70565cc0/log4j-core-2.20.0.jar:/home/dheeyantra/.gradle/caches/modules-2/files-2.1/org.apache.logging.log4j/log4j-api/2.20.0/1fe6082e660daf07c689a89c94dc0f49c26b44bb/log4j-api-2.20.0.jar:/home/dheeyantra/Projects/djl/engines/mxnet/mxnet-engine/build/libs/mxnet-engine-0.24.0-SNAPSHOT.jar:/home/dheeyantra/Projects/djl/engines/pytorch/pytorch-engine/build/libs/pytorch-engine-0.24.0-SNAPSHOT.jar:/home/dheeyantra/Projects/djl/engines/tensorflow/tensorflow-engine/build/libs/tensorflow-engine-0.24.0-SNAPSHOT.jar:/home/dheeyantra/Projects/djl/api/build/libs/api-0.24.0-SNAPSHOT.jar:/home/dheeyantra/.gradle/caches/modules-2/files-2.1/org.testng/testng/7.8.0/90ff6902a350432ce23ef209b2f109bcf587069c/testng-7.8.0.jar:/home/dheeyantra/.gradle/caches/modules-2/files-2.1/org.slf4j/slf4j-api/1.7.36/6c62681a2f655b49963a5983b8b0950a6120ae14/slf4j-api-1.7.36.jar:/home/dheeyantra/.gradle/caches/modules-2/files-2.1/org.apache.commons/commons-csv/1.10.0/8669bee353424c3223c93723291b5c3753260c1c/commons-csv-1.10.0.jar:/home/dheeyantra/.gradle/caches/modules-2/files-2.1/ml.dmlc/xgboost4j_2.12/1.7.5/5f074b329677d4e23222668597eb82de4b845a4b/xgboost4j_2.12-1.7.5.jar:/home/dheeyantra/.gradle/caches/modules-2/files-2.1/commons-logging/commons-logging/1.2/4bfc12adfe4842bf07b657f0369c4cb522955686/commons-logging-1.2.jar:/home/dheeyantra/.gradle/caches/modules-2/files-2.1/com.microsoft.ml.lightgbm/lightgbmlib/3.2.110/f6c85e5d7cc44d49c4544240ea5c96004680007b/lightgbmlib-3.2.110.jar:/home/dheeyantra/.gradle/caches/modules-2/files-2.1/com.microsoft.onnxruntime/onnxruntime/1.15.1/8e2c81b478bb0fc4dffc794fe71d5a99a372ce88/onnxruntime-1.15.1.jar:/home/dheeyantra/.gradle/caches/modules-2/files-2.1/com.google.code.gson/gson/2.10.1/b3add478d4382b78ea20b1671390a858002feb6c/gson-2.10.1.jar:/home/dheeyantra/.gradle/caches/modules-2/files-2.1/net.java.dev.jna/jna/5.13.0/1200e7ebeedbe0d10062093f32925a912020e747/jna-5.13.0.jar:/home/dheeyantra/.gradle/caches/modules-2/files-2.1/org.apache.commons/commons-compress/1.23.0/4af2060ea9b0c8b74f1854c6cafe4d43cfc161fc/commons-compress-1.23.0.jar:/home/dheeyantra/.gradle/caches/modules-2/files-2.1/com.beust/jcommander/1.82/a7c5fef184d238065de38f81bbc6ee50cca2e21/jcommander-1.82.jar:/home/dheeyantra/.gradle/caches/modules-2/files-2.1/org.webjars/jquery/3.6.1/d08df6250157cd2db3d9b01b11b76e9b7225083a/jquery-3.6.1.jar:/home/dheeyantra/Projects/djl/engines/tensorflow/tensorflow-api/build/libs/tensorflow-api-0.24.0-SNAPSHOT.jar:/home/dheeyantra/.gradle/caches/modules-2/files-2.1/org.tensorflow/tensorflow-core-api/0.5.0/6dfb7f13a9d96e6c4bd0705f122bd00d3b596b0d/tensorflow-core-api-0.5.0.jar:/home/dheeyantra/.gradle/caches/modules-2/files-2.1/org.bytedeco/javacpp/1.5.9/bee92b783ea619381df7577527f8739f778cf2f6/javacpp-1.5.9.jar:/home/dheeyantra/.gradle/caches/modules-2/files-2.1/com.google.protobuf/protobuf-java/3.23.3/6ea96f2109fb6cf8f827aa58eebf784c4708d01f/protobuf-java-3.23.3.jar:/home/dheeyantra/.gradle/caches/modules-2/files-2.1/org.tensorflow/ndarray/0.4.0/7ab74f002dbec93944b7feb38de013afe8d4e8de/ndarray-0.4.0.jar
java.vm.vendor: Ubuntu
sun.arch.data.model: 64
user.variant: 
java.vendor.url: https://ubuntu.com/
user.timezone: Asia/Kolkata
os.name: Linux
java.vm.specification.version: 11
sun.java.launcher: SUN_STANDARD
user.country: US
sun.boot.library.path: /usr/lib/jvm/java-11-openjdk-amd64/lib:/usr/lib/jvm/java-11-openjdk-amd64/lib
sun.java.command: ai.djl.integration.util.DebugEnvironment
jdk.debug: release
sun.cpu.endian: little
user.home: /home/dheeyantra
org.gradle.appname: gradlew
user.language: en
java.specification.vendor: Oracle Corporation
java.version.date: 2023-07-18
java.home: /usr/lib/jvm/java-11-openjdk-amd64
ai.djl.logging.level: debug
org.gradle.internal.http.connectionTimeout: 60000
file.separator: /
java.vm.compressedOopsMode: Zero based
line.separator: 

java.specification.name: Java Platform API Specification
java.vm.specification.vendor: Oracle Corporation
java.awt.graphicsenv: sun.awt.X11GraphicsEnvironment
sun.management.compiler: HotSpot 64-Bit Tiered Compilers
java.runtime.version: 11.0.20+8-post-Ubuntu-1ubuntu123.04
user.name: dheeyantra
path.separator: :
os.version: 6.2.0-27-generic
java.runtime.name: OpenJDK Runtime Environment
file.encoding: UTF-8
java.vm.name: OpenJDK 64-Bit Server VM
java.vendor.url.bug: https://bugs.launchpad.net/ubuntu/+source/openjdk-lts
java.io.tmpdir: /tmp
org.gradle.internal.http.socketTimeout: 120000
java.version: 11.0.20
user.dir: /home/dheeyantra/Projects/djl/integration
os.arch: amd64
java.vm.specification.name: Java Virtual Machine Specification
java.awt.printerjob: sun.print.PSPrinterJob
sun.os.patch.level: unknown
java.library.path: :/usr/local/cuda/lib64:/usr/local/cuda/extras/CUPTI/lib64:/usr/java/packages/lib:/usr/lib/x86_64-linux-gnu/jni:/lib/x86_64-linux-gnu:/usr/lib/x86_64-linux-gnu:/usr/lib/jni:/lib:/usr/lib
java.vm.info: mixed mode, sharing
java.vendor: Ubuntu
java.vm.version: 11.0.20+8-post-Ubuntu-1ubuntu123.04
java.specification.maintenance.version: 2
sun.io.unicode.encoding: UnicodeLittle
library.jansi.path: /home/dheeyantra/.gradle/native/jansi/1.18/linux64
java.class.version: 55.0
org.gradle.internal.publish.checksums.insecure: true

--------- Environment Variables ---------
PATH: /usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/snap/bin:/usr/local/cuda/bin
XAUTHORITY: /run/user/1000/gdm/Xauthority
LESSCLOSE: /usr/bin/lesspipe %s %s
XMODIFIERS: @im=ibus
XDG_DATA_DIRS: /usr/share/ubuntu:/usr/share/gnome:/usr/local/share/:/usr/share/:/var/lib/snapd/desktop
GDMSESSION: ubuntu
QT_IM_MODULE: ibus
TERM: xterm-256color
XDG_CONFIG_DIRS: /etc/xdg/xdg-ubuntu:/etc/xdg
GNOME_TERMINAL_SERVICE: :1.99
LANG: en_US.UTF-8
DBUS_SESSION_BUS_ADDRESS: unix:path=/run/user/1000/bus
XDG_SESSION_TYPE: x11
XDG_CURRENT_DESKTOP: ubuntu:GNOME
DISPLAY: :1
COLORTERM: truecolor
LD_LIBRARY_PATH: :/usr/local/cuda/lib64:/usr/local/cuda/extras/CUPTI/lib64
SESSION_MANAGER: local/EDGE-E01-BLR-01:@/tmp/.ICE-unix/1856,unix/EDGE-E01-BLR-01:/tmp/.ICE-unix/1856
USERNAME: dheeyantra
LOGNAME: dheeyantra
PWD: /home/dheeyantra/Projects/djl
XDG_SESSION_CLASS: user
_: ./gradlew
SHELL: /bin/bash
LESSOPEN: | /usr/bin/lesspipe %s
GPG_AGENT_INFO: /run/user/1000/gnupg/S.gpg-agent:0:1
DESKTOP_SESSION: ubuntu
OLDPWD: /home/dheeyantra/Projects/djl
USER: dheeyantra
XDG_MENU_PREFIX: gnome-
VTE_VERSION: 7200
QT_ACCESSIBILITY: 1
WINDOWPATH: 2
GNOME_DESKTOP_SESSION_ID: this-is-deprecated
SSH_AUTH_SOCK: /run/user/1000/keyring/ssh
GNOME_TERMINAL_SCREEN: /org/gnome/Terminal/screen/f5715672_9fe2_4064_95aa_f4586816c935
GTK_MODULES: gail:atk-bridge
CUDA_HOME: /usr/local/cuda
DEBUGINFOD_URLS: https://debuginfod.ubuntu.com 
GNOME_SHELL_SESSION_MODE: ubuntu
SYSTEMD_EXEC_PID: 3128
LS_COLORS: rs=0:di=01;34:ln=01;36:mh=00:pi=40;33:so=01;35:do=01;35:bd=40;33;01:cd=40;33;01:or=40;31;01:mi=00:su=37;41:sg=30;43:ca=00:tw=30;42:ow=34;42:st=37;44:ex=01;32:*.tar=01;31:*.tgz=01;31:*.arc=01;31:*.arj=01;31:*.taz=01;31:*.lha=01;31:*.lz4=01;31:*.lzh=01;31:*.lzma=01;31:*.tlz=01;31:*.txz=01;31:*.tzo=01;31:*.t7z=01;31:*.zip=01;31:*.z=01;31:*.dz=01;31:*.gz=01;31:*.lrz=01;31:*.lz=01;31:*.lzo=01;31:*.xz=01;31:*.zst=01;31:*.tzst=01;31:*.bz2=01;31:*.bz=01;31:*.tbz=01;31:*.tbz2=01;31:*.tz=01;31:*.deb=01;31:*.rpm=01;31:*.jar=01;31:*.war=01;31:*.ear=01;31:*.sar=01;31:*.rar=01;31:*.alz=01;31:*.ace=01;31:*.zoo=01;31:*.cpio=01;31:*.7z=01;31:*.rz=01;31:*.cab=01;31:*.wim=01;31:*.swm=01;31:*.dwm=01;31:*.esd=01;31:*.avif=01;35:*.jpg=01;35:*.jpeg=01;35:*.mjpg=01;35:*.mjpeg=01;35:*.gif=01;35:*.bmp=01;35:*.pbm=01;35:*.pgm=01;35:*.ppm=01;35:*.tga=01;35:*.xbm=01;35:*.xpm=01;35:*.tif=01;35:*.tiff=01;35:*.png=01;35:*.svg=01;35:*.svgz=01;35:*.mng=01;35:*.pcx=01;35:*.mov=01;35:*.mpg=01;35:*.mpeg=01;35:*.m2v=01;35:*.mkv=01;35:*.webm=01;35:*.webp=01;35:*.ogm=01;35:*.mp4=01;35:*.m4v=01;35:*.mp4v=01;35:*.vob=01;35:*.qt=01;35:*.nuv=01;35:*.wmv=01;35:*.asf=01;35:*.rm=01;35:*.rmvb=01;35:*.flc=01;35:*.avi=01;35:*.fli=01;35:*.flv=01;35:*.gl=01;35:*.dl=01;35:*.xcf=01;35:*.xwd=01;35:*.yuv=01;35:*.cgm=01;35:*.emf=01;35:*.ogv=01;35:*.ogx=01;35:*.aac=00;36:*.au=00;36:*.flac=00;36:*.m4a=00;36:*.mid=00;36:*.midi=00;36:*.mka=00;36:*.mp3=00;36:*.mpc=00;36:*.ogg=00;36:*.ra=00;36:*.wav=00;36:*.oga=00;36:*.opus=00;36:*.spx=00;36:*.xspf=00;36:*~=00;90:*#=00;90:*.bak=00;90:*.old=00;90:*.orig=00;90:*.part=00;90:*.rej=00;90:*.swp=00;90:*.tmp=00;90:*.dpkg-dist=00;90:*.dpkg-old=00;90:*.ucf-dist=00;90:*.ucf-new=00;90:*.ucf-old=00;90:*.rpmnew=00;90:*.rpmorig=00;90:*.rpmsave=00;90:
XDG_SESSION_DESKTOP: ubuntu
XDG_RUNTIME_DIR: /run/user/1000
SSH_AGENT_LAUNCHER: openssh
HOME: /home/dheeyantra
SHLVL: 1

-------------- Directories --------------
temp directory: /tmp
DJL cache directory: /home/dheeyantra/.djl.ai
Engine cache directory: /home/dheeyantra/.djl.ai

------------------ CUDA -----------------
GPU Count: 1
CUDA: 122
ARCH: 86
GPU(0) memory used: 1567948800 bytes

----------------- Engines ---------------
DJL version: 0.24.0-SNAPSHOT
[WARN ] - No matching cuda flavor for linux found: cu122mkl/sm_86.
[DEBUG] - Loading mxnet library from: /home/dheeyantra/.djl.ai/mxnet/1.9.1-mkl-linux-x86_64/libmxnet.so
[WARN ] - No matching cuda flavor for linux found: cu122mkl/sm_86.
Default Engine: MXNet:1.9.0, capabilities: [
        CPU_SSE,
        SIGNAL_HANDLER,
        LAPACK,
        BLAS_OPEN,
        CPU_SSE2,
        DIST_KVSTORE,
        CPU_SSE3,
        OPENMP,
        OPENCV,
        MKLDNN,
]
MXNet Library: /home/dheeyantra/.djl.ai/mxnet/1.9.1-mkl-linux-x86_64/libmxnet.so
Default Device: cpu()
PyTorch: 2
MXNet: 0
XGBoost: 10
LightGBM: 10
OnnxRuntime: 10
TensorFlow: 3

--------------- Hardware --------------
Available processors (cores): 12
Byte Order: LITTLE_ENDIAN
Free memory (bytes): 1043216632
Maximum memory (bytes): 16844324864
Total memory available to JVM (bytes): 1061158912
Heap committed: 1061158912
Heap nonCommitted: 31064064
GCC: 
gcc (Ubuntu 12.3.0-1ubuntu1~23.04) 12.3.0
Copyright (C) 2022 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

```

**Comment by technologistkj** [comment]
This fork can be used to replicate the issue: https://github.com/technologistkj/djl.git

Please run the class SpeechToTextGeneration (from the package ai.djl.examples.inference.whisper) on a GPU & cuda enabled system.

**Comment by frankfliu** [comment]
I created a PR to address ` Method 'forward' is not defined.` error: https://github.com/deepjavalibrary/djl/pull/2751

But the following error is due the the model. The model python code hard coded current device, and traced torchscript model contains device information. If you want to run the model on GPU(1), you must trace the model on GPU(1).

```
RuntimeError: Expected all tensors to be on the same device, but found at least two devices, cuda:0 and cpu! (when checking argument for argument index in method wrapper_CUDA__index_select)
```

I created another PR to force use CPU for whisper model: https://github.com/deepjavalibrary/djl/pull/2752
